### DISCUSSION TITLE
If the pick() can support rotate camera

### DISCUSSION THREAD

**Discussion by fanbinnet** [google/filament#5346] [Q&A]
Before I rotate the camera the callback of pick is right, But When I change the camera position, the callback result is wrong for some renderable.
If I need do some other for changing camera position ?

Thanks a lot for any help!

**Comment by romainguy** [comment]
It shouldn't matter, pick() renders the scene from the camera's viewpoint. @pixelflinger any idea?

**Reply by fanbinnet** [reply]
Below scene I can get right pick result
![u0](https://user-images.githubusercontent.com/3060688/159412318-6e8030df-31a0-4404-82d9-e8e6a73ca63f.JPG)

But I can't get right result after change camera:
![u1](https://user-images.githubusercontent.com/3060688/159412409-4e207b8b-6b79-4d6e-81e1-84c4f242efdf.JPG)
```
        cameraSphericalUtils.jumpToAngle(30f, -30f)
        cameraSphericalUtils.getLookAt(eyePos, target, upward)

    public void jumpToAngle( float thetaDegree,float phiDegree) {
        double theta = Math.toRadians(thetaDegree);
        double phi = Math.toRadians(phiDegree);
        float x = (float) (Math.sin(theta) * Math.cos(phi));
        float y = (float) Math.sin(phi);
        float z = (float) (Math.cos(theta) * Math.cos(phi));
        Float3 coords = new Float3(x, y, z);
        mEye = mPivot.plus(coords.times((float) Math.abs(mRadius)));
        mTarget = mEye.plus(coords.times(mFlipped ? 1.0f : -1.0f));
    }
```
The entity id my renderable is start with "39", but When I click some ball I get result "37".

**Reply by pixelflinger** [reply]
I think for this one it would be better to file a bug (issue) and provide a sample if possible.

**Reply by zhangbo-0213** [reply]
when I use pick() function, the id of renderable in result I got is always the same value(is 38 in my scene), whichever I click the renderable in scene, is there something wrong that I called function?
